google/resource_runtimeconfig_config.go

```diff
@@ -0,0 +1,148 @@
+package google
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/runtimeconfig/v1beta1"
+)
+
+var runtimeConfigFullName *regexp.Regexp
+
+func init() {
+	runtimeConfigFullName = regexp.MustCompile("^projects/([^/]+)/configs/(.+)$")
+}
+
+func resourceRuntimeconfigConfig() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceRuntimeconfigConfigCreate,
+		Read:   resourceRuntimeconfigConfigRead,
+		Update: resourceRuntimeconfigConfigUpdate,
+		Delete: resourceRuntimeconfigConfigDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateGCPName,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceRuntimeconfigConfigCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	name := d.Get("name").(string)
+	fullName := resourceRuntimeconfigFullName(project, name)
+	runtimeConfig := runtimeconfig.RuntimeConfig{
+		Name: fullName,
+	}
+
+	if val, ok := d.GetOk("description"); ok {
+		runtimeConfig.Description = val.(string)
+	}
+
+	_, err = config.clientRuntimeconfig.Projects.Configs.Create("projects/"+project, &runtimeConfig).Do()
+
+	if err != nil {
+		return err
+	}
+	d.SetId(fullName)
+
+	return nil
+}
+
+func resourceRuntimeconfigConfigRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	fullName := d.Id()
+	runConfig, err := config.clientRuntimeconfig.Projects.Configs.Get(fullName).Do()
+	if err != nil {
+		return err
+	}
+
+	project, name, err := resourceRuntimeconfigParseFullName(runConfig.Name)
+	if err != nil {
+		return err
+	}
+	// Check to see if project matches our current defined value - if it doesn't, we'll explicitly set it
+	curProject, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+	if project != curProject {
+		d.Set("project", project)
+	}
+
+	d.Set("name", name)
+	d.Set("description", runConfig.Description)
+
+	return nil
+}
+
+func resourceRuntimeconfigConfigUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	fullName := d.Id()
+	runtimeConfig := runtimeconfig.RuntimeConfig{
+		Name: fullName,
+	}
+	// Update the description (currently its the only thing that could have changed)
+	if v, ok := d.GetOk("description"); ok {
```

Comment:
I believe this logic is wrong. If you set the description, apply, then unset it, and re-apply, then the description won't be erased. You only change the description if it is set but unsetting it is also something the user might want to do. using hasChange instead like suggested in my previous comment will solve this.